src/Tools/SchemaTool.php

```diff
@@ -337,7 +337,8 @@ public function getSchemaFromMetadata(array $classes): Schema
 
             if (isset($class->table['uniqueConstraints'])) {
                 foreach ($class->table['uniqueConstraints'] as $indexName => $indexData) {
-                    $uniqIndex = new Index($indexName, $this->getIndexColumns($class, $indexData), true, false, [], $indexData['options'] ?? []);
+                    $uniqIndexName = is_numeric($indexName) ? null : $indexName;
+                    $uniqIndex     = new Index($uniqIndexName, $this->getIndexColumns($class, $indexData), true, false, [], $indexData['options'] ?? []);
```

Comment:
ORM 3.0 still supports DBAL 3.6 and up, where `Doctrine\DBAL\Schema\Index`  can't have null as an index name. A nullable name was introduced in DBAL 4.0. It looks like the `Index` instance is only created to be used in `$tableIndex->isFulfilledBy($uniqIndex)` a few lines later.

Reply:
Hmm, yes, you are right. As I understand it, this check was added as part of issue 1375. Under the current circumstances, there is another quick fix that I see: to compare with existing indexes, use an index with an empty name if the array key is a number. Given that the index name is not used for comparison and is declared as an empty string by default, this looks like it would work. Tests also confirm this for dbal versions 3.8 and 4.0. However, when adding a unique index, passing the index name as null seems to be fine, as long as the index name is empty. On the other hand, it doesn't seem very nice. What do you think about that ?

Reply:
I don't expect that the name will be used in a check, because the focus is on involved columns to prevent duplicates. It would be the same having two unique indices with `null`. Please try your suggested approach on this.

Reply:
Fine, I've made a few minor edits.